File: Common/src/main/java/com/cerbon/talk_balloons/config/TBConfig.java
package com.cerbon.talk_balloons.config;

import com.cerbon.talk_balloons.util.TBConstants;
import me.shedaniel.autoconfig.ConfigData;
import me.shedaniel.autoconfig.annotation.Config;
import me.shedaniel.cloth.clothconfig.shadowed.blue.endless.jankson.Comment;

@Config(name = TBConstants.MOD_ID)
public class TBConfig implements ConfigData {

    public float balloonsHeightOffset = 0.9f;
    public int distanceBetweenBalloons = 3;

    public int maxBalloons = 7;
    public int minBalloonWidth = 13;
    public int maxBalloonWidth = 180;

    @Comment("In seconds")
    public int balloonAge = 15;

    public int textColor = 1315860;

    public boolean showOwnBalloon = true;
}

File: common/src/main/java/com/cerbon/talk_balloons/config/TBConfig.kt
package com.cerbon.talk_balloons.config

object TBConfig {
    var balloonsHeightOffset: Float = 0.9f
    var distanceBetweenBalloons: Int = 3

    var maxBalloons: Int = 7

    var minBalloonWidth: Int = 13
    var maxBalloonWidth: Int = 180

    var balloonPadding: Int = 1
    var balloonAge: Int = 15
    var balloonStyle: BalloonStyle = BalloonStyle.ROUNDED

    var textColor: Int = 0x141414 // RGB-encoded
    var balloonTint: Int = 0xF1F6F8 // RGB-encoded

    var showOwnBalloon: Boolean = true
    var onlyDisplayBalloons: Boolean = false
}
